roles/remote-user/tasks/main.yml

```diff
@@ -0,0 +1,5 @@
+---
+- name: Determine whether to connect as root or admin_user
+  local_action: shell ssh -o PasswordAuthentication=no root@{{ groups.web.0 }} "echo root" || echo {{ admin_user }}
+  changed_when: false
+  register: remote_user
```

Comment:
I've adjusted this to check all hosts instead of just the first. It sets the `ansible_ssh_user` fact for each host. Now the playbook can handle variability in whether or not hosts permit `root` login.